plugins/docker_binary_builder/lib/samson_docker_binary_builder/samson_plugin.rb

```diff
@@ -0,0 +1,66 @@
+module SamsonDockerBinaryBuilder
+  class Engine < Rails::Engine
+  end
+end
+
+def build_docker_binaries(dir, project, reference, output)
+  output.puts 'Now building the build container...'
+  Docker::Image.build_from_dir(dir, { 'dockerfile' => 'Dockerfile.build', 't' => "#{project.send(:permalink_base)}_build:#{reference}" })
```

Comment:
Yup, code is a mess, was going to tidy it up :)

And yea, .tar.tar is a result of container.copy() automatically tar'ing it
for you! :/


On Mon, Aug 24, 2015 at 4:05 AM, Roman Sandler <notifications@github.com>
wrote:

> In
> plugins/docker_binary_builder/lib/samson_docker_binary_builder/samson_plugin.rb
> <https://github.com/zendesk/samson/pull/538#discussion_r37716507>:
>
> > @@ -0,0 +1,66 @@
> > +module SamsonDockerBinaryBuilder
> > +  class Engine < Rails::Engine
> > +  end
> > +end
> > +
> > +def build_docker_binaries(dir, project, reference, output)
> > +  output.puts 'Now building the build container...'
> > +  Docker::Image.build_from_dir(dir, { 'dockerfile' => 'Dockerfile.build', 't' => "#{project.send(:permalink_base)}_build:#{reference}" })
>
> I guess everything should be conditional on this dockerfile existing?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendesk/samson/pull/538/files#r37716507>.
>